tests/cpp_unit_tests/test_tap_position_optimizer.cpp

```diff
@@ -0,0 +1,61 @@
+// SPDX-FileCopyrightText: Contributors to the Power Grid Model project <powergridmodel@lfenergy.org>
+//
+// SPDX-License-Identifier: MPL-2.0
+
+#include <power_grid_model/optimizer/tap_position_optimizer.hpp>
+
+#include <boost/graph/graph_utility.hpp>
+
+#include <doctest/doctest.h>
+
+namespace pgm_tap = power_grid_model::optimizer::tap_position_optimizer;
+
+namespace power_grid_model {
+namespace {} // namespace
+
+TEST_CASE("Test Transformer ranking") {
+    // The grid
+
+    // Subcases
+    SUBCASE("Building the graph") {
+        // graph creation
+    }
+
+    std::vector<pgm_tap::TrafoGraphEdge> edge_array;
+    edge_array.push_back(pgm_tap::TrafoGraphEdge({{0, 1}, 1}));
+    edge_array.push_back(pgm_tap::TrafoGraphEdge({{1, 2}, 2}));
+
+    pgm_tap::TransformerGraph g;
+    //{boost::edges_are_unsorted_multi_pass, edge_array.cbegin(), edge_array.cend(), 3};
+
+    SUBCASE("Process edge weights") { auto const res = get_edge_weights(g); }
+
+    SUBCASE("Dijkstra shortest path") {
+        std::vector<pgm_tap::EdgeWeight> edge_weight(3, pgm_tap::infty);
+        std::vector<Idx2D> edge_pos(3);
+        // process_edges_dijkstra(0, edge_weight, edge_pos, g);
+    }
```

Comment:
i don't necessarily think that there's there's a need to determine shortest path itself. just a check that the input and output of the ranking is correct. all the rest is implementation detail (if you prefer, you can put it in `namespace detail`)